### ISSUE TITLE
Multi-stage build fails when CLI invoked with --pull

### ISSUE THREAD

**Issue by DavidObando** [moby/moby#34512]
**Description**

The docker daemon will try to go to the docker registry and pull newer images during build operations when the docker CLI is invoked with the `--pull` flag. When using a multi-stage build, the daemon is attempting to fetch the intermediate stages from the public registry, and fails with the following message:
```
invalid from flag value vpbuilder: repository sha256 not found: does not exist or no pull access
```

**Steps to reproduce the issue:**
You can reproduce with the following simple files:
### Dockerfile
```Dockerfile
FROM golang:1.8 as vpbuilder
WORKDIR /go/src/vp
COPY . .
RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -v -a -o vp .

FROM alpine:3.4
WORKDIR /vp/
COPY --from=vpbuilder /go/src/vp/vp .
ENTRYPOINT ./vp
```

### main.go
```go
package main

import (
        "fmt"
)

func main() {
        fmt.Println("Hello, bug report!")
}
```

Then, execute the following command:
```bash
docker build --pull -t vp1 .
```

**Describe the results you received:**
The build fails, and I see the following message:
```
invalid from flag value vpbuilder: repository sha256 not found: does not exist or no pull access
```

**Describe the results you expected:**
The build should succeed just as when I call docker build without the `--pull` flag.

**Output of `docker version`:**

```
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:17:04 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:15:57 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 11
Server Version: 17.06.0-ce
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 16
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: cfb82a876ecc11b5ca0977d1733adbe58599088a
runc version: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 4.8.0-59-generic
Operating System: Ubuntu 16.10
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 11.72GiB
Name: daobando-u1
ID: S4FY:DU3O:TGZJ:XGVH:CQAO:TAW7:C3YS:KGYU:E2VD:RTWO:ZKXN:7JQP
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by jeremyhuiskamp** [comment]
Looks like this has been addressed with https://github.com/moby/moby/pull/33735 but not yet released.

**Comment by thaJeztah** [comment]
Thanks for reporting

@jeremyhuiskamp is correct, this is a duplicate of https://github.com/moby/moby/issues/33946 and https://github.com/docker/for-mac/issues/1751, and is fixed on "master" through https://github.com/moby/moby/pull/33735, which will be included in the upcoming Docker CE 17.06.1 release.

I'm closing this issue because this is already fixed in this repository, but feel free to comment after I closed
